op-node/rollup/driver/state.go

```diff
@@ -327,7 +327,7 @@ func (s *Driver) eventLoop() {
 			s.metrics.SetDerivationIdle(false)
 			s.idleDerivation = false
 			s.log.Debug("Derivation process step", "onto_origin", s.derivation.Origin(), "attempts", stepAttempts)
-			err := s.derivation.Step(context.Background())
+			err := s.derivation.Step(ctx)
```

Comment:
The cancellation of this `ctx` is synchronous with the function call, and will thus never happen. We could maybe put a timeout on the `Step`, but shouldn't set a step-timeout until the find-sync-start is no longer part of the stepping.

Reply:
yea, this is true. Maybe it should be `context.TODO()` to make it clear it's not being used for now

Reply:
`context.TODO()` might trigger some lint. Would just keep it as-is until we refactor the find-sync-start bit to be separate.